doc/source/dev/data_types.md

```diff
@@ -0,0 +1,312 @@
+```xml
+<command>someTool.sh ${os.path.join( input1.extra_files_path, 'results.txt' )} ${os.path.join( input1.extra_files_path, 'results.dat' )} $output1</command>
+```
+
+
+### Advanced Topics: `Rgenetics` Example 
+
+`Rgenetics` datatypes are defined as composite datatypes. The tools in this package rely heavily on the contents of a filename for analysis and reporting. In this case it is desirable for the filenames of the components to vary slightly, but maintain a common form. To do this, we use a special metadata parameter that can only be set at dataset creation (i.e. upload). This example uses the metadata parameter **base_name** to specify part of the components' filenames.
+```python
+class Lped(Text):
+    MetadataElement( name="base_name", desc="base name for all transformed versions of this genetic dataset", default="galaxy", readonly=True, set_in_upload=True)
+
+
+    composite_type = 'auto_primary_file'
+    file_ext="lped"
+    allow_datatype_change = False
+
+
+    def __init__ ( self, **kwd ):
+        Rgenetics. __init__ ( self, **kwd )
+        self.add_composite_file( '%s.ped', description = 'Pedigree File', substitute_name_with_metadata = 'base_name', is_binary = True )
+        self.add_composite_file( '%s.map', description = 'Map File', substitute_name_with_metadata = 'base_name', is_binary = True )
+
+
+    def generate_primary_file( self, dataset = None ):
+        rval = ['<html><head><title>Files for Composite Dataset (%s)</title></head><p/>This composite dataset is composed of the following files:<p/><ul>' % ( self.file_ext )]
+        for composite_name, composite_file in self.get_composite_files( dataset = dataset ).iteritems():
+            opt_text = ''
+            if composite_file.optional:
+                opt_text = ' (optional)'
+            rval.append( '<li><a href="%s">%s</a>%s' % ( composite_name, composite_name, opt_text ) )
+        rval.append( '</ul></html>' )
+        return "\n".join( rval )
+```
+
+The file specified as `%s.ped` is found at `os.path.join( input1.extra_files_path, '%s.ped' % input1.metadata.base_name )`.
+
+It should be noted that changing the datatype of datasets which use this substitution method will cause an error if the metadata parameter 'base_name' does not exist in a datatype that the dataset is set to. This is because the value within 'base_name' will be lost -- if the datatype is set back to the original datatype, the default metadata value will be used and the filenames might not match the basename.
+
+To prevent this from occurring, set `allow_datatype_change` to `False`. The dataset's datatype will not be able to be changed.
+
+### Future Directions
+
+This approach was implemented to be backwards compatible with existing Rgenetics datasets/tools (implemented before composite datatypes existed). Different approaches should be considered. Suggestions and comments are welcome.
+
+## Galaxy Tool Shed - Data Types
```

Comment:
Given the discussion above, should all the following just be removed? It can stay on the Hub.

Reply:
Given that it is a feature, I'd opt to keep it in, I'm adding the 'completely new data type' part as well.

Reply:
I added that section that they are deprecated and shouldn't be used.